pkg/services/accesscontrol/resourcepermissions/api_adapter.go

```diff
@@ -0,0 +1,168 @@
+package resourcepermissions
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/authlib/types"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+
+	dashboardv1 "github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v1beta1"
+	folderv1 "github.com/grafana/grafana/apps/folder/pkg/apis/folder/v1beta1"
+	iamv0 "github.com/grafana/grafana/apps/iam/pkg/apis/iam/v0alpha1"
+	"github.com/grafana/grafana/pkg/api/dtos"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/team"
+	"github.com/grafana/grafana/pkg/services/user"
+)
+
+func (a *api) getDynamicClient(ctx context.Context) (dynamic.Interface, error) {
+	if a.restConfigProvider == nil {
+		return nil, fmt.Errorf("k8s rest config provider not available")
```

Comment:
Let's define that variable globally and catch it so we log it with a lower level for `teams`, `datasources` and `service accounts` permissions. 
`Teams` being a special case as it won't go the resource permission apis; but rather the team bindings ones.